utils/src/main/java/com/cloud/utils/nio/NioConnection.java

```diff
@@ -202,57 +237,64 @@ protected void accept(final SelectionKey key) throws IOException {
-            sslEngine = Link.initServerSSLEngine(caService, socketChannel.getRemoteAddress().toString());
-            sslEngine.setUseClientMode(false);
-            sslEngine.setEnabledProtocols(SSLUtils.getSupportedProtocols(sslEngine.getEnabledProtocols()));
             final NioConnection nioConnection = this;
-            _sslHandshakeExecutor.submit(new Runnable() {
-                @Override
-                public void run() {
-                    _selector.wakeup();
-                    try {
-                        sslEngine.beginHandshake();
-                        if (!Link.doHandshake(socketChannel, sslEngine)) {
-                            throw new IOException("SSL handshake timed out with " + socketChannel.getRemoteAddress());
-                        }
-                        if (logger.isTraceEnabled()) {
-                            logger.trace("SSL: Handshake done");
-                        }
-                        final InetSocketAddress saddr = (InetSocketAddress)socket.getRemoteSocketAddress();
-                        final Link link = new Link(saddr, nioConnection);
-                        link.setSSLEngine(sslEngine);
-                        link.setKey(socketChannel.register(key.selector(), SelectionKey.OP_READ, link));
-                        final Task task = _factory.create(Task.Type.CONNECT, link, null);
-                        registerLink(saddr, link);
-                        _executor.submit(task);
-                    } catch (IOException e) {
-                        if (logger.isTraceEnabled()) {
-                            logger.trace("Connection closed due to failure: " + e.getMessage());
-                        }
-                        closeAutoCloseable(socket, "accepting socket");
-                        closeAutoCloseable(socketChannel, "accepting socketChannel");
-                    } finally {
-                        _selector.wakeup();
+            _sslHandshakeExecutor.submit(() -> {
+                final InetSocketAddress socketAddress = (InetSocketAddress)socket.getRemoteSocketAddress();
+                activeAcceptConnections.incrementAndGet();
+                long startTime = System.currentTimeMillis();
+                _selector.wakeup();
+                try {
+                    final SSLEngine sslEngine = Link.initServerSSLEngine(caService, socketChannel.getRemoteAddress().toString());
+                    sslEngine.setUseClientMode(false);
+                    sslEngine.setEnabledProtocols(SSLUtils.getSupportedProtocols(sslEngine.getEnabledProtocols()));
+                    sslEngine.beginHandshake();
+                    if (!Link.doHandshake(socketChannel, sslEngine, getSslHandshakeTimeout())) {
+                        throw new IOException("SSL handshake timed out with " + socketAddress);
+                    }
+                    if (logger.isTraceEnabled()) {
+                        logger.trace("SSL: Handshake done");
                     }
```

Comment:
No need for a guard here, but if you really want to, you could use lazy logging.
```suggestion
                    logger.trace("SSL: Handshake done");
```